webpack/assets/javascripts/react_app/components/factCharts/FactChart.test.js

```diff
@@ -0,0 +1,134 @@
+import { shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import React from 'react';
+import FactChart from '.';
+import Store from '../../redux';
+
+import { getChartData, showModal, closeModal } from './FactChartActions';
+
+import { ajaxRequestAction } from '../../redux/actions/common/index';
+
+import reducer from './FactChartReducer';
+import * as types from './FactChartConstants';
+
+import {
+  initialState,
+  modalOpenState,
+  modalSuccessState,
+  modalLoadingState,
+  modalErrorState,
+  chartDataValues,
+} from './FactChart.fixtures';
+
+describe('factCharts', () => {
+  it('should render closed', () => {
+    const wrapper = shallow(
+      <FactChart data={{ id: 1, title: 'test title' }} store={Store} />
+    );
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should render open', () => {
+    const wrapper = shallow(
+      <FactChart
+        data={{ id: 1, title: 'test title' }}
+        modalToDisplay
+        store={Store}
+      />
+    );
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should open modal after click', () => {
+    const props = { onClick: jest.fn(), data: { id: 1 } };
+    const wrapper = shallow(<FactChart store={Store} {...props} />);
+
+    wrapper.simulate('click');
+    expect(props.onClick).toBeCalled();
+  });
+});
+
+jest.unmock('./FactChartActions');
+jest.mock('../../redux/actions/common');
+describe('factCharts actions', () => {
+  it('getChartData should call ajaxRequestAction with url and id', () => {
+    const id = 1;
+    const url = 'url';
+    const dispatch = jest.fn();
+    const expectedParams = {
+      dispatch,
+      failedAction: types.FACT_CHART_DATA_FAILURE,
+      item: { id },
+      requestAction: types.FACT_CHART_DATA_REQUEST,
+      successAction: types.FACT_CHART_DATA_SUCCESS,
+      url,
+    };
+    const dispatcher = getChartData(url, id);
+
+    dispatcher(dispatch);
+    expect(ajaxRequestAction).toBeCalledWith(expectedParams);
+  });
+
+  it('should open modal', () => {
+    const id = 1;
+    const title = 'test title';
+    const expectedResults = {
+      type: types.OPEN_FACT_CHART_MODAL,
+      payload: { id, title },
+    };
+
+    expect(showModal(id, title)).toEqual(expectedResults);
+  });
+
+  it('should close modal', () => {
+    const id = 1;
+    const expectedResults = {
+      type: types.CLOSE_FACT_CHART_MODAL,
+      payload: { id },
+    };
+
+    expect(closeModal(id)).toEqual(expectedResults);
+  });
+});
+
+describe('factCharts reducers', () => {
```

Comment:
can you move this to a reducer test file? there is a  helper function as well (`testReducerSnapshotWithFixtures`)

Reply:
moved